File: pset6/caesar/caesar.py
from cs50 import get_int, get_string
import sys
import argparse

"""
    // Autor: Luisángel Martín Marcia Palma
    // PSET 6 caesar.py
"""

def caesarcypher(k):

    k = int(k)

    pt = get_string("plaintext:")
    counter = int(0)
    n = len(pt)
    print("ciphertext: ", end="")
    encription = ''
    for i in range(0, n, 1):
        c = ord(pt[i]) + k % 26
        if(pt[i].isalpha()):
            if(pt[i].isupper()):
                if(c > 90):
                  encription += chr(c - 26)
                else:
                     encription += chr(c)
            elif pt[i].islower():
                if(c > 122):
                    encription += chr(c - 26)
                else:
                    encription += chr(c)
        else:
            encription += pt[i]

    print(encription)
    return

def main():
    if sys.argv[1].isdigit() == False:
        raise Exception('Usage: ./caesar key')

    key = int(sys.argv[1])
    caesarcypher(key)
    return

if __name__ == '__main__':
    main()

File: pset2/caesar.c
#include<stdio.h>
#include<stdlib.h>
#include<cs50.h>
#include<ctype.h>
#include<string.h>

// Author: Luisángel Martín Marcia Palma
// PSET 2 caesar.c

// Global variables
string p;

// Calling functions
int caesarcypher(int k);

int main (int argc, string argv[])
{
    //argv[0] --> nombre de programa
    if (argc != 2)
    {
        printf ("Usage: ./caesar key");
        return 1;
    }

    // Defining variables
    int key = atoi (argv[1]);
    caesarcypher(key);
    return 0;
}

// caesarcypher function
int caesarcypher(int k)
{
    p = get_string("plaintext:");
    int c = 0;
    printf ("ciphertext:");
    for (int i = 0, n = strlen (p); i < n; i++)
    {
        if (isupper (p[i]))
        {
            c = (p[i]+k-65)%26+65;
            printf ("%c", p[i] = c);
        }
        else
        if (islower (p[i]))
        {
            c = (p[i]+k-97)%26+97;
            printf ("%c", p[i] = c);
        }
        else
            printf ("%c", p[i]);

    }
    printf ("\n");
    return 0;
}